tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_advertiser.c

```diff
@@ -262,6 +381,86 @@ static void test_tx_cb_multi(void)
 	PASS();
 }
 
+static void test_tx_proxy_mixin(void)
+{
+	static struct bt_mesh_prov prov = {
+		.uuid = test_prov_uuid,
+	};
+	int err;
+
+	/* Initialize mesh stack and enable pb gatt bearer to emit beacons. */
+	bt_mesh_device_setup(&prov, &comp);
+	err = bt_mesh_prov_enable(BT_MESH_PROV_GATT);
+	ASSERT_OK(err, "Failed to enable GATT provisioner");
+
+	/* Start pb gatt beacons with 100 msec interval */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provision device under test");
+	/* Provision dut and start gatt proxy beacons. */
+	bt_mesh_provision(test_net_key, 0, 0, 0, adv_cfg.addr, adv_cfg.dev_key);
+
+	/* Start proxy beacons with 1 sec interval */
+	k_sleep(K_MSEC(6000));
+
+	/* Send adv data */
+	struct net_buf *buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, BT_MESH_LOCAL_ADV,
+			BT_MESH_TRANSMIT(5, 20), K_NO_WAIT);
+	net_buf_add_mem(buf, txt_msg, sizeof(txt_msg));
+	bt_mesh_adv_send(buf, NULL, NULL);
+	k_sleep(K_MSEC(150));
+
+	/* Start proxy beacons again */
+	k_sleep(K_MSEC(6000));
+
+	PASS();
+}
+
+static void test_rx_proxy_mixin(void)
+{
+	static struct bt_mesh_test_gatt gattparam = {
+		.transmits = 1500 / 100,
+		.interval = 100,
+		.tolerance = 20,
+		.service = MESH_PROV_SERVICE
+	};
+
+	bt_init();
+
+	/* Scan pb gatt beacons. */
+	gatt_param = &gattparam;
```

Comment:
It should be separate pointer since it is used in scanner callback. No way to pass it there other than global value. It doesn't matter this is pointer or just variable. I don't think it has any sense, advantage or disadvantage to change it on variable or make it as incoming parameter for `rx_gatt_beacons`.

Which more meaningful name do you assume? When I gave name I assumed gatt session parameter.

Reply:
Then either of this is possible and should make the code more readable:
- Access `gatt_param` directly in the test function. This is kind of already done in few lines below, but then I don't see why it can't be done initially.
- Instantiate `gattparam` in the test function and pass it by value or pointer to `rx_gatt_beacons`. Then it creates a clear connection between this structure and the function, saves few minutes of developers figuring out the relation of this code and makes life simpler;